src/createServer.js

```diff
@@ -1,3 +1,66 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    const { url } = req;
+    const [text, queryString] = url.substring(1).split('?');
+    const params = new URLSearchParams(queryString);
+
+    const toCase = params.get('toCase');
+
+    const errors = [];
+
+    if (!text) {
+      errors.push({
+        message: 'Text to convert is required. Correct request is: '
+        + '"/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!toCase) {
+      errors.push({
+        message: '"toCase" query param is required. Correct request is: '
+        + '"/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    const supportedCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+    if (!supportedCases.includes(toCase) && toCase) {
+      res.writeHead(400, { 'Content-Type': 'application/json' });
+
+      errors.push({
+        message: 'This case is not supported. Available cases:'
+        + ' SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+      });
+    }
+
+    if (errors.length !== 0) {
+      res.writeHead(400, { 'Content-Type': 'application/json' });
+
+      res.end(
+        JSON.stringify({ errors }),
+      );
+
+      return;
+    }
+
+    const result = convertToCase(text, toCase);
+
+    res.writeHead(200, { 'Content-Type': 'application/json' });
+
+    res.end(
+      JSON.stringify({
+        originalCase: result.originalCase,
+        targetCase: toCase,
+        originalText: text,
+        convertedText: result.convertedText,
+      }),
+    );
```

Comment:
Can be simplified a little, [writeHead](https://nodejs.org/api/http.html#responsewriteheadstatuscode-statusmessage-headers) already return `res` so you can chain it.
```suggestion
    res
      .writeHead(200, { 'Content-Type': 'application/json' })
      .end(
        JSON.stringify({
          ...result,
          targetCase: toCase,
          originalText: text,
        }),
    );
```